src/mui/list/List.js

```diff
@@ -102,10 +106,9 @@ export class List extends Component {
         return this.props.location.pathname;
     }
 
-    refresh = (event) => {
+    refresh = event => {
         event.stopPropagation();
-        this.fullRefresh = true;
-        this.updateData();
```

Comment:
I know the data list rerenders, but the question is: do references in the datalist get rerendered as well, and only *after* the main list is rerendered? You can check this by watching the HTTP requests done by the app.